cognite/client/_api/simulators/routines.py

```diff
@@ -0,0 +1,189 @@
+from __future__ import annotations
+
+from collections.abc import Iterator, Sequence
+from typing import TYPE_CHECKING, Any, overload
+
+from cognite.client._api_client import APIClient
+from cognite.client._constants import DEFAULT_LIMIT_READ
+from cognite.client.data_classes._base import CogniteFilter
+from cognite.client.data_classes.simulators.filters import CreatedTimeSort, SimulatorRoutinesFilter
+from cognite.client.data_classes.simulators.routines import (
+    SimulatorRoutine,
+    SimulatorRoutineList,
+    SimulatorRoutineWrite,
+)
+from cognite.client.utils._experimental import FeaturePreviewWarning
+from cognite.client.utils._identifier import IdentifierSequence
+from cognite.client.utils._validation import assert_type
+from cognite.client.utils.useful_types import SequenceNotStr
+
+if TYPE_CHECKING:
+    from cognite.client import ClientConfig, CogniteClient
+
+
+class SimulatorRoutinesAPI(APIClient):
+    _RESOURCE_PATH = "/simulators/routines"
+
+    def __init__(self, config: ClientConfig, api_version: str | None, cognite_client: CogniteClient) -> None:
+        super().__init__(config, api_version, cognite_client)
+        self._warning = FeaturePreviewWarning(
+            api_maturity="General Availability", sdk_maturity="alpha", feature_name="Simulators"
+        )
+
+    def __iter__(self) -> Iterator[SimulatorRoutine]:
+        """Iterate over simulator routines
+
+        Fetches simulator routines as they are iterated over, so you keep a limited number of simulator routines in memory.
+
+        Returns:
+            Iterator[SimulatorRoutine]: yields Simulator routines one by one.
+        """
+        return self()
+
+    @overload
+    def __call__(
+        self, chunk_size: int, filter: SimulatorRoutinesFilter | None = None, limit: int | None = None
+    ) -> Iterator[SimulatorRoutineList]: ...
+
+    @overload
+    def __call__(
+        self, chunk_size: None = None, filter: SimulatorRoutinesFilter | None = None, limit: int | None = None
+    ) -> Iterator[SimulatorRoutine]: ...
+
+    def __call__(
+        self, chunk_size: int | None = None, filter: SimulatorRoutinesFilter | None = None, limit: int | None = None
```

Comment:
Same here as the list method. Don't use the `SumulatorRoutinesFilter` instead add the arguments directly as parameters.